### PR TITLE
MODLOGIN-62 SenderService implementation

### PR DESCRIPTION
**Pull request by dmytrotur**
MODSENDER-2

 * Endpoint **POST /message-delivery** was implemented
 * _DeliveryVerticle_ class was implemented to register delivery channels
 * _DeliveryChannel_ interface was implemented by _EmailDeliveryChannel_
 * Email delivery channel was registered with name 'email'
 * new functionality was covered with API tests in _MessageDeliveryTest_

Links to issue:
https://issues.folio.org/browse/MODSENDER-2

### CODE REVIEW

**Review by dcrossleyau — COMMENTED** [review]
I cannot assist with technical code review, just some setup stuff.

* Thanks for doing the initial schema property descriptions. That helps everyone.
I see that you have some more expressive descriptions. Good.
There is further encouragement in a new document at https://dev.folio.org/guides/describe-schema/

* Regarding copies of schema files from other modules:
We did used to have those in raml-util together with shared RAMLs, but that became unwieldy and was stopped. I do not know if there is higher-level discussion somewhere about it.
(Not expecting this to be resolved in this PR.)

* Consistency with the spelling: it is "FOLIO" not "Folio".

File: src/main/java/org/folio/rest/impl/MessageDeliveryImpl.java [hunk 1]
```diff
@@ -4,20 +4,45 @@
...
+import org.folio.rest.model.OkapiHeaders;
+import org.folio.rest.tools.utils.TenantTool;
+import org.folio.sender.SenderService;
+import org.folio.sender.SenderServiceImpl;
+import org.folio.sender.util.ExceptionHelper;
 
 import javax.ws.rs.core.Response;
 import java.util.Map;
 
 public class MessageDeliveryImpl implements MessageDelivery {
```
Comment (opochernin) [inline src/main/java/org/folio/rest/impl/MessageDeliveryImpl.java:20]: please add logs to this class

Comment (dmytrotur) [reply src/main/java/org/folio/rest/impl/MessageDeliveryImpl.java:20]: Exceptions are logged in ExceptionHelper#mapExceptionToResponse. I'll probably rename it to something like "handleException" to show method's purpose.

Comment (opochernin) [reply src/main/java/org/folio/rest/impl/MessageDeliveryImpl.java:20]: got it, thanks for the clarification

**Review by opochernin — APPROVED** [review]
Looks good.

**Comment by dmytrotur** [comment]
@dcrossleyau thanks for your review.
I fixed FOLIO spelling and added more descriptions where possible.

**Pull request merged**